sdk/typescript/src/client/ika-transaction.ts

```diff
@@ -0,0 +1,1722 @@
+	 *
+	 * @param params - The parameters for accepting the encrypted user share
+	 * @param params.dWallet - The DWallet object to accept the share for
+	 * @param params.encryptedUserSecretKeyShareId - The ID of the encrypted user secret key share
+	 * @returns Promise resolving to the updated IkaTransaction instance
+	 * @throws {Error} If user share encryption keys are not set
+	 */
+	async acceptEncryptedUserShare({
+		dWallet,
+		encryptedUserSecretKeyShareId,
+	}: {
+		dWallet: DWallet;
+		encryptedUserSecretKeyShareId: string;
+	}) {
+		if (!this.userShareEncryptionKeys) {
+			throw new Error('User share encryption keys are not set');
+		}
+
+		coordinatorTx.acceptEncryptedUserShare(
+			this.ikaClient.ikaConfig,
+			this.getCoordinatorObjectRef(),
+			dWallet.id.id,
+			encryptedUserSecretKeyShareId,
+			await this.userShareEncryptionKeys.getUserOutputSignature(dWallet),
+			this.transaction,
+		);
+
+		return this;
+	}
+
+	/**
+	 * Register an encryption key for the current user on the specified curve.
+	 * This allows the user to participate in encrypted operations on the network.
+	 *
+	 * @param params - The parameters for registering the encryption key
+	 * @param params.curve - The elliptic curve identifier to register the key for
+	 * @returns Promise resolving to the updated IkaTransaction instance
+	 * @throws {Error} If user share encryption keys are not set
+	 */
+	async registerEncryptionKey({ curve }: { curve: Curve }) {
+		if (!this.userShareEncryptionKeys) {
+			throw new Error('User share encryption keys are not set');
+		}
+
+		coordinatorTx.registerEncryptionKeyTx(
+			this.ikaClient.ikaConfig,
+			this.getCoordinatorObjectRef(),
+			curve,
+			this.userShareEncryptionKeys.encryptionKey,
+			await this.userShareEncryptionKeys.getEncryptionKeySignature(),
+			this.userShareEncryptionKeys.getSigningPublicKeyBytes(),
+			this.transaction,
+		);
+
+		return this;
+	}
+
+	/**
+	 * Make the DWallet user secret key shares public, allowing them to be used without decryption.
+	 * This is useful for scenarios where the secret share can be publicly accessible.
+	 *
+	 * @param params - The parameters for making the secret key shares public
+	 * @param params.dWallet - The DWallet to make the shares public for
+	 * @param params.secretShare - The secret share data to make public
+	 * @param params.ikaCoin - The IKA coin object to use for transaction fees
+	 * @param params.suiCoin - The SUI coin object to use for gas fees
+	 * @returns The updated IkaTransaction instance
+	 */
+	makeDWalletUserSecretKeySharesPublic({
+		dWallet,
+		secretShare,
+		ikaCoin,
+		suiCoin,
+	}: {
+		dWallet: DWallet;
+		secretShare: Uint8Array;
+		ikaCoin: TransactionObjectArgument;
+		suiCoin: TransactionObjectArgument;
+	}) {
+		coordinatorTx.requestMakeDwalletUserSecretKeySharesPublic(
+			this.ikaClient.ikaConfig,
+			this.getCoordinatorObjectRef(),
+			dWallet.id.id,
+			secretShare,
```

Comment:
there's an inconsistency here: `requestMakeDwalletUserSecretKeySharesPublic()` expects `publicUserSecretKeyShares` i.e. an array, this one gives a single value?

Reply:
I get that its a byte array, but is it a single secret share or an array of secret shares? it should be the latter given the function you call, and if its the former then you should first wrap it inside an array with just this single item i believe